### PR TITLE
[Task 1899887] refactor the logic to generate staging folder

### PR DESCRIPTION
**Pull request by andxu**
[AB#1899887](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1899887)

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/refelection/RefelectionFunctionAdaptor.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import javax.annotation.Nonnull;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class RefelectionFunctionAdaptor {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/refelection/RefelectionFunctionAdaptor.java]: typo

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/core/FunctionClass.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+package com.microsoft.azure.toolkit.lib.appservice.function.core;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.List;
+
+public class FunctionClass implements IAnnotatable {
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/core/FunctionClass.java:12]: FunctionAnnotationClass or AnnotationClass

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/refelection/MavenGradleFunctionStagingContributor.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.*;
+import java.util.stream.Collectors;
+
+class MavenGradleFunctionStagingContributor {
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/refelection/MavenGradleFunctionStagingContributor.java:35]: may be merged with `FunctionProject`

**Review by wangmingliang-ms — APPROVED** [review]
remember to resolve the comments in later PRs.

**Comment by andxu** [comment]
New pr https://github.com/microsoft/azure-maven-plugins/pull/1864 addressed the comments.

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/core/FunctionClass.java (hunk 2 above)
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/core/FunctionClass.java:12]: New pr https://github.com/microsoft/azure-maven-plugins/pull/1864 addressed the comments.

**Pull request merged**